misc/bash/completion.sh

```diff
@@ -0,0 +1,19 @@
+function _complete_nix {
```

Comment:
Would it make sense to do the same for zsh? https://github.com/spwhitt/nix-zsh-completions

Reply:
If somebody wants to contribute that code, sure.

Reply:
cc @spwhitt

Reply:
If we have something to share I support doing so, however I believe Eelco wants completions that are generated by the Nix command, and our current completions are not. Also @olejorgenb and @hedning are in charge of the project as far as I'm concerned, whatever they want to do is fine with me.